tests/unit/sagemaker/jumpstart/test_model.py

```diff
@@ -0,0 +1,277 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+from typing import Optional
+from unittest import mock
+import unittest
+import pytest
+
+from sagemaker.jumpstart.model import JumpStartModel
+from sagemaker.predictor import Predictor
+
+from tests.unit.sagemaker.jumpstart.utils import get_special_model_spec, overwrite_dictionary
+
+
+class ModelTest(unittest.TestCase):
+
+    execution_role = "fake role! do not use!"
+    region = "us-west-2"
+
```

Comment:
could you please add test cases to ensure that:
- that the image is retrieved with `INFERENCE` scope
- if the user does not pass an instance type, the default instance is retrieved before retrieving the image
- if the user passes an instance type, the image is retrieved with that instance
- that intelligent defaults for `vpc`, each `kms` key, `role` and `network_isolation` are correctly handled

If any of these tests are not possible at the top level, make sure you add coverage for them in the kwargs utils.